src/api/middleware/validators/offer.js

```diff
@@ -130,6 +130,160 @@ const create = useExpressValidators([
         .isArray(),
 ]);
 
+const edit = useExpressValidators([
+    body("_id", ValidationReasons.DEFAULT)
+        .exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(async (_id, { req }) => {
+            try {
+                const offer = await (new OfferService()).getOfferById(_id, req.user);
+                if (!offer) throw new Error(ValidationReasons.OFFER_NOT_FOUND(_id));
+                if (req.user?.company.toString() !== offer.owner.toString()) throw new Error(ValidationReasons.NOT_OFFER_OWNER(_id));
+            } catch (_e) {
+                // Also catches any fail to the DB
+                throw new Error(_e);
+            }
+
+            return true;
+        }),
```

Comment:
This should be another middleware, as you said, something like `hasPermissionsToEdit` which verifies if the logged in user is the company owner, or if it is admin or god (these 2 are missing it seems)

Reply:
So should I create a custom function `hasPermissionsToEdit`. But should I do it in `id` aswell? Like `param("offerId).custom(hasPermissionsToEdit)`

Reply:
No, it should be another middleware (that runs before the request middleware), similar to the isCompanyOrGod, for example.

You'd have something like isCompanyOrGod => isEditableOffer => hasPermissionsToEdit => validators.edit

Reply:
I see, will do!